lib/developer_portal/app/controllers/developer_portal/admin/applications_controller.rb

```diff
@@ -113,10 +113,12 @@ def applications
   end
 
   def service
-    @service ||= if not site_account.multiservice?
-                   site_account.services.default
-                 elsif service_id = params[:service_id]
-                   site_account.services.find_by_id_or_system_name(service_id)
+    @service ||= if (service_id = params[:service_id])
+                   services.find_by(id: service_id) || services.find_by(system_name: service_id)
```

Comment:
Avoid doing 2 requests:

```suggestion
                   services.where(id: service_id).or(system_name: service_id).first
```